lib/src/runner/live_suite_controller.dart

```diff
@@ -106,8 +106,12 @@ class LiveSuiteController {
   /// is `true` (the default), the test is put into [passed] if it succeeds.
   /// Otherwise, it's removed from [liveTests] entirely.
   ///
+  /// Likewise, if [countFailure] is `true` (the default), the test is put into
+  /// [failed] if it fails. Otherwise, it's removed from [liveTests] entirely.
```

Comment:
I played around with trying to move the logic into the Invoker but was unfortunately unsuccessful. Ensuring the errors are printed, dealing with the timeout, and correctly setting the state resulted in a lot of additional logic. I quite like the idea of simply cloning the test and reusing all the existing logic. That being said I understand your concerns so lets chat in person and see if the difficulty of moving the logic into the Invoker was simply a result of my lack of experience with this code base.